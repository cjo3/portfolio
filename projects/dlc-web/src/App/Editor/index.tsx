import { Grid, Button, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'
import { FileLoader } from './FileLoader'
import { SheetSelector } from './SheetSelector'
import { TransformResults } from './TransformResults'
import { TransformSettings } from './TransformSettings'
import { Link } from 'react-router-dom'
import { switchLinkRoutePath } from '@cjo3/shared/react/helpers'

const useStyles = makeStyles(theme => ({
  Editor_header: {
    width: '100%',
    padding: theme.custom.setSpace('lg'),
    backgroundImage:
      'url(https://eskipaper.com/images/high-tech-wallpapers-3.jpg)',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    color: theme.palette.grey[50],
    textShadow: theme.custom.textShadow,
    textAlign: 'center'
  },
  Editor_header_instructionsButton: {
    marginTop: theme.custom.setSpace('sm')
  }
}))

export const Editor: React.FC = (): JSX.Element => {
  const classes = useStyles()

  return (
    <Grid container>
      <Grid item component="header" className={classes.Editor_header}>
        <Typography variant="h1" align="center">
          Editor
        </Typography>
        <Typography variant="body1" align="center">
          Labore labore ut est sit sanctus stet et. Eos lorem ipsum consetetur
          magna est voluptua. Lorem justo dolor eirmod est est aliquyam. Dolor
          sit kasd nonumy lorem tempor dolor no duo et, invidunt gubergren rebum
          at rebum, et nonumy lorem lorem diam ipsum, sadipscing diam voluptua
          accusam sit sit, dolor.
        </Typography>
        <Button
          type="button"
          variant="contained"
          color="secondary"
          className={classes.Editor_header_instructionsButton}
          href={switchLinkRoutePath(
            '/editor/instructions',
            `${process.env.APP_ROOT_PATH}/editor/instructions`
          )}>
          Instructions
        </Button>
      </Grid>
      <FileLoader />
      <SheetSelector />
      <TransformSettings />
      <TransformResults />
    </Grid>
  )
}
