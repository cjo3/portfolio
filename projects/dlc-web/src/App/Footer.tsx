import { PrivacyPolicy } from '@cjo3/shared/react/components/PrivacyPolicy'
import { TermsAndConditions } from '@cjo3/shared/react/components/TermsAndConditions'
import { ImageHandler } from '@cjo3/shared/react/components/ImageHandler'
import { switchLinkRoutePath } from '@cjo3/shared/react/helpers'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import GavelIcon from '@material-ui/icons/Gavel'
import VpnLockIcon from '@material-ui/icons/VpnLock'
import WebIcon from '@material-ui/icons/Web'
import clsx from 'clsx'
import React, { useState } from 'react'
import { ProfilePic } from '../../assets/'

const useStyles = makeStyles(theme => ({
  Footer_container: {
    padding: theme.custom.setSpace('sm'),
    background: theme.custom.setLinearGradient(
      180,
      theme.palette.grey[900],
      theme.palette.grey[800]
    )
  },
  Footer_linkButton: {
    ...theme.custom.setFlex('row', 'flext-start', 'center'),
    'marginTop': theme.custom.setSpace(),
    '&:first-child': {
      marginTop: 0
    }
  },
  Footer_linkButtonIcon: {
    fontSize: theme.typography.fontSize * 1.5,
    marginRight: theme.custom.setSpace() / 2,
    position: 'relative',
    top: -2
  },
  Footer_badgeTitleImageContainer: {
    ...theme.custom.setFlex('row', 'flex-end'),
    [theme.breakpoints.down('xs')]: {
      ...theme.custom.setFlex('row', 'flex-start'),
      margin: `${theme.custom.setSpace('sm')}px 0 0 0`
    },
    '&:hover': {
      cursor: 'pointer'
    }
  },
  Footer_badgeProfilePic: {
    'order': 2,
    'borderRadius': theme.custom.setSpace() / 4,
    [theme.breakpoints.down('xs')]: {
      order: 1
    },
    'transition': 'all 250ms ease-out',
    '&:hover': {
      boxShadow: theme.custom.boxShadow
    }
  },
  Footer_badgeTitle: {
    'textAlign': 'right',
    'order': 1,
    'marginRight': theme.custom.setSpace(),
    [theme.breakpoints.down('xs')]: {
      order: 2,
      textAlign: 'left',
      marginLeft: theme.custom.setSpace(),
      marginRight: 0
    },
    'transition': 'all 250ms ease-out',
    '&:hover': {
      color: theme.palette.primary[400]
    }
  },
  Footer_badgeSubtitle: {
    color: theme.palette.grey[400],
    textAlign: 'right',
    [theme.breakpoints.down('xs')]: {
      textAlign: 'left'
    }
  },
  Footer_copyright: {
    marginTop: theme.custom.setSpace('sm'),
    textAlign: 'right',
    [theme.breakpoints.down('xs')]: {
      textAlign: 'left'
    }
  }
}))

interface Props {
  style: string
}

export const Footer: React.FC<Props> = ({ style }): JSX.Element => {
  const classes = useStyles()

  const [TAndCOpen, setTAndCOpen] = useState<boolean>(false)

  const [PPOpen, setPPOpen] = useState<boolean>(false)

  const openTAndCHandler = (): void => setTAndCOpen(true)

  const closeTAndCHandler = (): void => setTAndCOpen(false)

  const openPPHandler = (): void => setPPOpen(true)

  const closePPHandler = (): void => setPPOpen(false)

  const badgeClickHandler = (): void => {
    if (window)
      window.open(process.env.SITE_URL, '_blank') ||
        window.location.replace(process.env.SITE_URL)
  }

  const homeLinkHandler = (): void => {
    if (window)
      window.location.replace(
        switchLinkRoutePath('/', process.env.APP_ROOT_PATH)
      )
  }

  return (
    <Grid
      container
      component="footer"
      className={clsx(style, classes.Footer_container)}>
      {TAndCOpen && (
        <TermsAndConditions
          open={TAndCOpen}
          closeHandler={closeTAndCHandler}
          siteName={process.env.SITE_NAME}
          siteUrl={process.env.SITE_URL}
          siteContactEmail={process.env.SITE_CONTACT_EMAIL}
        />
      )}
      {PPOpen && (
        <PrivacyPolicy
          open={PPOpen}
          closeHandler={closePPHandler}
          siteName={process.env.SITE_NAME}
          siteUrl={process.env.SITE_URL}
          siteContactEmail={process.env.SITE_CONTACT_EMAIL}
        />
      )}
      <Grid item xs={12} sm={6}>
        <Grid
          container
          justify="flex-start"
          alignItems="flex-start"
          direction="column">
          <Button
            type="button"
            color="primary"
            onClick={homeLinkHandler}
            className={classes.Footer_linkButton}>
            <WebIcon className={classes.Footer_linkButtonIcon} />
            Home
          </Button>
          <Button
            type="button"
            color="primary"
            className={classes.Footer_linkButton}
            onClick={openTAndCHandler}>
            <GavelIcon className={classes.Footer_linkButtonIcon} />
            Terms &amp; Conditions
          </Button>
          <Button
            type="button"
            color="primary"
            className={classes.Footer_linkButton}
            onClick={openPPHandler}>
            <VpnLockIcon className={classes.Footer_linkButtonIcon} />
            Privacy Policy
          </Button>
        </Grid>
      </Grid>
      <Grid item xs={12} sm={6}>
        <Grid
          container
          className={classes.Footer_badgeTitleImageContainer}
          onClick={badgeClickHandler}>
          <Typography
            variant="h5"
            color="primary"
            className={classes.Footer_badgeTitle}>
            Need Help with JavaScript
            <br />
            App Development?
          </Typography>
          <ImageHandler
            asset={ProfilePic}
            styleClass={classes.Footer_badgeProfilePic}
          />
        </Grid>
        <Typography variant="body1" className={classes.Footer_badgeSubtitle}>
          Available for Hire!
        </Typography>
        <Typography
          variant="body1"
          color="primary"
          className={classes.Footer_copyright}>
          &copy; 2020 {process.env.SITE_NAME}. All rights reserved.
        </Typography>
      </Grid>
    </Grid>
  )
}
