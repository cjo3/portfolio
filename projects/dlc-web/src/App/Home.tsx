import { Header } from '@cjo3/shared/react/components/Header'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'
import { HomeHeader } from '../assets'

const useStyles = makeStyles(theme => ({
  Home_container: {
    height: '100%'
  }
}))

export const Home: React.FC = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container className={classes.Home_container}>
      <Header
        title={process.env.APP_NAME}
        subTitle="Quickly change detection limit values"
        bgColor="theme.palette.primary.main"
        bgUrl={HomeHeader.paths[0]}
        buttonHref="/editor"
        buttonLabel="Start"
      />
    </Grid>
  )
}
