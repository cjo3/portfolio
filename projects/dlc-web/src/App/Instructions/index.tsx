import React from 'react'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  Instructions_header: {
    padding: theme.custom.setSpace('sm')
  }
}))

interface Props {}

export const Instructions: React.FC<Props> = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container>
      <Grid item component="header" className={classes.Instructions_header}>
        <Typography variant="h1" align="center">
          Instructions
        </Typography>
        <Typography variant="body1" align="center">
          Labore labore ut est sit sanctus stet et. Eos lorem ipsum consetetur
          magna est voluptua. Lorem justo dolor eirmod est est aliquyam. Dolor
          sit kasd nonumy lorem tempor dolor no duo et, invidunt gubergren rebum
          at rebum, et nonumy lorem lorem diam ipsum, sadipscing diam voluptua
          accusam sit sit, dolor.
        </Typography>
      </Grid>
    </Grid>
  )
}
