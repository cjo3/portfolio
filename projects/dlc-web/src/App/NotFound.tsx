import { switchLinkRoutePath } from '@cjo3/shared/react/helpers'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'

const useStyles = makeStyles(theme => ({
  NotFound_contentContainer: {
    minHeight: '100%'
  },
  NotFound_header: {
    ...theme.custom.setFlex('column'),
    padding: theme.custom.setSpace('md'),
    backgroundImage:
      'url(https://image.freepik.com/free-photo/rendering-abstract-futuristic-background-with-glowing-neon-blue-orange-lights_181624-19807.jpg)',
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    color: theme.palette.grey[50]
  },
  NotFound_startButton: {
    marginTop: theme.custom.setSpace('sm')
  }
}))

export const NotFound: React.FC = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container className={classes.NotFound_contentContainer}>
      <Grid item xs={12} component="header" className={classes.NotFound_header}>
        <Typography variant="h1" align="center">
          Error
        </Typography>
        <Typography variant="body1" align="center">
          Sorry, something went wrong
        </Typography>
        <Button
          type="button"
          variant="contained"
          href={switchLinkRoutePath('/', process.env.APP_ROOT_PATH)}
          color="secondary"
          className={classes.NotFound_startButton}>
          Home
        </Button>
      </Grid>
    </Grid>
  )
}
