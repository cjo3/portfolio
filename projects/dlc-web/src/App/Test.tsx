import { processSheet } from '@cjo3/shared/logic/dlc'
import { sheetData as sheetDataMock } from '@cjo3/shared/react/mocks/dlc'
import { sortAddresses } from '@cjo3/shared/react/xlsx'
import { Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'

const useStyles = makeStyles(theme => ({
  padding: {
    padding: theme.custom.setSpace('sm')
  }
}))

export const Test: React.FC = (): JSX.Element => {
  const classes = useStyles()

  const values = {
    rangeStart: 'B17',
    rangeEnd: 'AM47',
    ulTrigger: '<',
    ulTriggerZero: 'Rock',
    ulTransform: 'halve',
    olTrigger: '>',
    olTransform: 'halve'
  }

  const result = processSheet(sheetDataMock, values)

  console.log('%c result', 'color: yellow; font-size: large', result)

  return (
    <Grid container>
      <Grid item className={classes.padding}>
        <Typography variant="body1" align="center"></Typography>
      </Grid>
    </Grid>
  )
}
