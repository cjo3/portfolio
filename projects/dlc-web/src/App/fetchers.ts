import { post } from 'axios'
import { addAuthHeaderToOptions } from '@cjo3/shared/security/authToken'

const otherHeaders = {
  headers: {
    accept: 'application/json'
  }
}

export const makePreOrder = async () => {
  let apiUrl = 'http://localhost:2000'

  if (process.env.NODE_ENV === 'production')
    apiUrl = process.env.API_URL + '/detection-limit-corrector'

  const res = await post(
    `${apiUrl}/pre-order`,
    null,
    addAuthHeaderToOptions(otherHeaders)
  )
  return res.data
}
