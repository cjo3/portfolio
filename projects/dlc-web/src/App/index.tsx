import { CssBaseline } from '@material-ui/core'
import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { Editor } from './Editor'
import { Home } from './Home'
import { NotFound } from './NotFound'
import { Footer } from './Footer'
import { Preview } from './Editor/Preview/'

export const App: React.FC = (): JSX.Element => (
  <CssBaseline>
    <Switch>
      <Route path="/" exact component={Home} />
      <Route path="/editor" exact component={Editor} />
      <Route path="/test" exact component={Preview} />
      <Route path="/*" component={NotFound} />
    </Switch>
    <Footer />
  </CssBaseline>
)
