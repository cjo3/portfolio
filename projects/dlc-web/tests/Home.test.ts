import { testRenderer, renderSnapshot } from './testRenderer'
import { Home } from '../src/App/Home'

describe('Home', () => {
  describe('render', () => {
    test('matches current snapshop', () => {
      const tree = renderSnapshot(Home, '/')
      expect(tree).toMatchSnapshot()
    })

    test('renders heading', () => {
      const { getByRole } = testRenderer('/', Home)
      const heading = getByRole('heading', { level: 1 })
      expect(heading).toBeInTheDocument()
    })

    test('renders copy', () => {
      const { getByText } = testRenderer('/', Home)
      const copy = getByText('consetetur', {
        selector: 'p',
        exact: false
      })
      expect(copy).toBeInTheDocument()
    })

    test('renders link to editor', () => {
      const { getByRole } = testRenderer('/', Home)
      const startLink = getByRole('link', { name: /start/i })
      expect(startLink).toHaveAttribute('href', '/editor')
    })
  })

  describe('events', () => {
    test('link clicks to editor', () => {
      const {
        getByRole,
        userEvent: { click }
      } = testRenderer('/')
      const startLink = getByRole('link', { name: /start/i })
      click(startLink)
      const heading = getByRole('heading', { level: 1 })
      expect(heading.innerHTML).toBe('Editor')
    })
  })
})
