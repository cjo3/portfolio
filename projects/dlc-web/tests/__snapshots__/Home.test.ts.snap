// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render matches current snapshop 1`] = `
<section
  className="MuiGrid-root makeStyles-section-1 MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      Nonumy magna lorem erat at eirmod et. Labore ut no diam accusam aliquyam amet sit, sanctus nonumy labore no sea aliquyam erat ea. Dolor sit diam accusam erat, kasd nonumy eos aliquyam invidunt. Vero tempor amet vero aliquyam invidunt consetetur est est sed. Invidunt diam sanctus sea amet invidunt.
    </p>
  </div>
  <a
    href="/editor"
  >
    Start
  </a>
</section>
`;
